extensions/notebook/src/book/bookTreeView.ts

```diff
@@ -0,0 +1,115 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+import * as path from 'path';
+import * as fs from 'fs';
+import * as yaml from 'js-yaml';
+import { BookTreeItem } from './bookTreeItem';
+import * as utils from '../common/utils';
+import * as nls from 'vscode-nls';
+const localize = nls.loadMessageBundle();
+
+
+export class BookTreeViewProvider implements vscode.TreeDataProvider<BookTreeItem> {
+
+	private _onDidChangeTreeData: vscode.EventEmitter<BookTreeItem | undefined> = new vscode.EventEmitter<BookTreeItem | undefined>();
+	readonly onDidChangeTreeData: vscode.Event<BookTreeItem | undefined> = this._onDidChangeTreeData.event;
+	private tableOfContentsPath: string[];
+
+	constructor(private workspaceRoot: string) {
+		if (workspaceRoot !== '') {
+			this.tableOfContentsPath = this.getTocFiles(this.workspaceRoot);
+			if (this.tableOfContentsPath === undefined || this.tableOfContentsPath.length === 0) {
+				vscode.commands.executeCommand('setContext', 'bookOpened', false);
+			} else {
+				vscode.commands.executeCommand('setContext', 'bookOpened', true);
+			}
+		}
+	}
+
+	private getTocFiles(dir: string): string[] {
+		let allFiles: string[] = [];
+		let files = fs.readdirSync(dir);
+		for (let i in files) {
+			let name = path.join(dir, files[i]);
+			if (fs.statSync(name).isDirectory()) {
+				allFiles = allFiles.concat(this.getTocFiles(name));
+			} else if (files[i] === 'toc.yml') {
+				allFiles.push(name);
+			}
+		}
+		return allFiles;
+	}
+
+	async openNotebook(resource: vscode.Uri): Promise<void> {
+		try {
+			let doc = await vscode.workspace.openTextDocument(resource);
+			vscode.window.showTextDocument(doc);
+		} catch (e) {
+			vscode.window.showErrorMessage(localize('openNotebookError', 'Open file {0} failed: {1}',
+				resource.fsPath,
+				e instanceof Error ? e.message : e));
+		}
+	}
+
+	getTreeItem(element: BookTreeItem): vscode.TreeItem {
```

Comment:
this seems like an unneeded method?  Is this to satisfy an interface requirement?

Reply:
Yep, it's needed for the TreeDataProvider interface